ffi/cpp/include/decimal.hpp

```diff
@@ -0,0 +1,71 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#ifndef ICU4X_DECIMAL_HPP
+#define ICU4X_DECIMAL_HPP
+
+#include <algorithm>
+#include <memory>
+
+#include "../../capi/include/decimal.h"
+#include "locale.hpp"
+#include "provider.hpp"
+#include "fixed_decimal.hpp"
+#include "utils.hpp"
+
+namespace icu4x {
+    struct ICU4XFixedDecimalFormatDeleter {
```

Comment:
nit: here and elsewhere, please don't indent for namespaces, https://google.github.io/styleguide/cppguide.html#Namespace_Formatting